File: code-examples/java-rs/indexes/multikey.java
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoCollection;
import com.mongodb.reactivestreams.client.MongoDatabase;
import org.bson.Document;

import static com.mongodb.client.model.Filters.eq;
import static com.mongodb.client.model.Projections.fields;
import static com.mongodb.client.model.Projections.include;
import static com.mongodb.client.model.Projections.excludeId;

public class MongoDBAccess {
    public static void main(String[] args) {
        MongoClient mongoClient = MongoClients.create("mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority");

        MongoDatabase database = mongoClient.getDatabase("sample_mflix");
        MongoCollection<Document> movies = database.getCollection("movies");

        movies.createIndex(new Document("cast", 1)).subscribe(indexName -> {
            System.out.println("Index created: " + indexName);

            Document query = eq("cast", "Viola Davis");
            Document projection = fields(include("cast", "title"), excludeId());

            movies.find(query)
                  .projection(projection)
                  .subscribe(movie -> System.out.println(movie.toJson()), 
                             error -> error.printStackTrace(), 
                             mongoClient::close);
        });
    }
}

File: code-examples/pymongo/indexes/multikey.py
# Import MongoClient from the PyMongo library
from pymongo import MongoClient

# Replace the placeholders with your credentials
uri = "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority"

client = MongoClient(uri)

# Connect to the database and the 'movies' collection
database = client['sample_mflix']
movies = database['movies']

# Create a multikey index on the 'cast' field
result = movies.create_index('cast')

# Print the result of creating the index
print(f'Index created: {result}')

# Query to find all movies featuring actor 'Viola Davis'
query = {"cast": "Viola Davis"}

# Projection to return only 'title' and 'cast', excluding '_id'
projection = {"_id": 0, "cast": 1, "title": 1}

# Perform the query with the filter and projection, and print the results
cursor = movies.find(query, projection)
for doc in cursor:
    print(doc)

# Close the connection
client.close()